python/vosk/aligner/forced_aligner.py

```diff
@@ -0,0 +1,48 @@
+from . import diff_align
+from . import text_processor
+from . import metasentence
+from . import multipass
+from .transcriber import Transcriber as transcriber
+from .transcription import Transcription
+
+class ForcedAligner():
+    '''Head class of the program which control all basic things, providing
+    language and acoustic models input args and getting results. ForcedAligner
+    is watching for aligning process(align/realign parts) allow to see
+    alignment results. Output word sequence contain whole information each
+    word(status, timings, etc).
+    '''
+    def __init__(self, transcript, model):
+        self.model = model
+        self.ks = transcript
+        self.ms = metasentence.MetaSentence(self.ks, self.model)
+        self.text = text_processor.process_text(self.ks, self.model)
+
+    def transcribe(self, wavfile, progress_cb=None, logging=None):
+        words = transcriber.transcribe(self.text, wavfile)
+
+        def unalign(words):
+            NFIA = len([X for X in words if (X.not_found_in_audio())])
+            NFIT = len([X for X in words if (X.not_found_in_transcript())])
+            amount = NFIA + NFIT
+            length = len(words)
+            return amount, length
+
+        # Align words
+        words = diff_align.align(words, self.ms)
+
+        # Perform a second-pass with unaligned words
+        if logging is not None:
+            amount, length = unalign(words)
+            logging.info("%d unaligned words (of %d)", amount, length)
+
+        if amount != 0:
```

Comment:
`amount` is unassigned if `logging is None`

Reply:
Hello, thanks for you help, I will fix it :)

Reply:
> Hello, thanks for you help, I will fix it :)

Hello did you have success fixing it?

Reply:
Hello, no for a while, but I hope to start it after finish my current project

Reply:
I returned to aligner project, need to rework code a bit

Reply:
Wonderful 👍 So excited, can't wait to try. @vadimdddd Do you an email or way I contact you to collaborate? Would love to share some thoughts and ideas.

Reply:
@CodeFusionFX, sure things - sieget4nk90@gmail.com